doc/developer-notes.md

```diff
@@ -460,7 +460,10 @@ other input.
   safely continue even if the assumption is violated. In debug builds it
   behaves like `Assert`/`assert` to notify developers and testers about
   nonfatal errors. In production it doesn't warn or log anything, though the
-  expression is always evaluated.
+  expression is always evaluated. However, if the compiler can prove that
+  an expression inside `Assume` is side-effect-free, it may optimize the call away,
```

Comment:
```suggestion
  an expression inside `Assume` is true, it _will_ have no affect,
```

Reply:
s/affect/effect

Reply:
will fix not when there is need to retouch.

Reply:
> If the expression inside Assume has a side-effect, but still can be proven to evaluate to true, the side effect will still remain.

I am assuming you are talking about the Macro then and not the C++ 23 [Assume](http://fiona.dmcs.p.lodz.pl/oopc/reference/en/cpp/language/attributes/assume.html) since that's not how the docs read.  The docs read that if it's true it has no effect: "If the expression would not evaluate to true at the place the assumption occurs, the behavior is undefined. Otherwise, the statement has no effect. "

Also, I think _may_ should be replaced with _will_ unless it's ambigous wether or not this will be otimized away depending on it's side-effect which it doesn't sound like it is.

Reply:
I just noticed that Sipa said "There is no guarantee that the compiler will do that." which I guess is why the _may_ exists here.

Reply:
> Yes, nothing in this PR is about the C++23 [[assume()]] attribute. I don't understand why you keep bringing it up.

The name collision had me confused